### DISCUSSION TITLE
✅ ✨ Introducing Elementor AI - for Images ✨

### DISCUSSION THREAD

**Discussion by avivu** [elementor/elementor#22744] [Announcement]
![AI_images_GH](https://github.com/elementor/elementor/assets/76733085/16cd8282-8756-4171-8f11-3272145ba60e)

**Hi Beta Testers** 👋

We are excited to reveal the next phase of Elementor AI, available in Elementor 3.14.0 - Beta 4. Now, you can use Elementor AI to create original images and enhance existing images on your website, without ever having to leave the Editor.

With Elementor AI for image generation, you can:

- **Bring your vision to life** - Simply describe any image you want to create in plain text, and let Elementor AI do the magic.
- **Customize** - Fine tune your images by adding elements, editing or extending them to match the aspect ratio you need for your website.
- **Get inspired** - View a gallery of premade AI-generated images and copy any of the prompts used to create the images in the gallery, modify it and create your own unique images.

Elementor AI is just getting started and more innovation is coming soon.

# Ignite Your Creativity with Elementor AI for Image Creation

Elementor 3.14.0 - Beta 4 introduces the first phase of Elementor AI’s Image Generator, which extends the previous, text generating capabilities, introduced in the previous version of Elementor. With this new capability, you can unleash your creativity, improve and accelerate your workflow by instantly generating any type of image for your website, directly from within the Editor. This means that you don’t need to go searching through millions of stock images to find the specific image you want to use, download it to your website, and then upload it into Elementor. With Elementor AI, you can create images, leverage the built-in prompt enhancer to elevate your prompts, and results, and use the generated images directly within your website.
 
With Elementor AI’s image generator, you can also leverage its editing and customization features to add elements to your image, update its aspect ratio, resolution, and expand an image beyond its original borders by providing a description of the changes you want to make.
 
Elementor AI’s Image Generator also includes a gallery of images created with AI that you can use on your website. All the images included in the gallery also include a prompt you can copy into the Image Generator, so you can modify it and create your own image.
 
In upcoming versions, Elementor will continue updating its AI capabilities to supercharge your workflow and websites.

#### How to Test It:

1. Create a new page and drag an image widget into the page.
2. Click on the “Create with AI” button in the widget panel.
3. If prompted, connect to your account by clicking the “Connect” button.
4. Accept the Terms and Conditions & Privacy Policy of Elementor AI and click “Get Started”.
5. You now have access to all currently available Elementor AI features (please note: you need to have Elementor 3.14.0 - beta 4  or higher in order to use the AI Images features).
6. Start testing the “Image generation” feature by writing prompts to generate images.
7. Use the “enhance prompt” button to create more elaborate prompts based on just a few keywords (e.g. “orange, fruit, bowl” + enhance prompt)
8. Continue by trying different image types and styles using the dropdown menus in the image generating modal.
9. Next try the “AI Image Editing” feature by clicking the “Edit” button (visible when hovering over an image), or by selecting an existing image widget in the canvas and clicking “Edit with AI” in the widgets panel.
10. Start by using the “Expand Image” tool: choose the aspect ratio and then use the size slider and the drag and drop functionality to place the image in the appropriate position. Fill in the prompt if you want to generate something specific, or just leave it blank to let the AI do expand it on its own (tip: expand works best on rich images)
11. Continue by going into “Edit” mode and clicking the “Generative fill” button. Use the canvas brush to paint over the area you want to edit, enter your prompt and hit “generate.”
12. Proceed to test the “Resize” tool by going into “Edit” mode and selecting a larger size for the selected image.
13. Finally, go into “Edit” mode and try the “Variations” tool to generate multiple variations from a given image.
14. Try different prompts, configurations and make sure everything works as expected.

-----

## Make An Impact: Share Feedback Before Features Are Released

We’re always looking to get to know Elementor customers on a deeper level. If you’re interested in meeting with our team and sharing your insights, feedback, and feature requests, you’re welcome to sign up for some quality time with us [here](https://elementor.com/pages/joinresearch/). Talk soon!

### How to Install the Beta Version
If you still haven't enabled beta testing in Elementor, please follow these steps:

- Go to Elementor > Tools > Versions
- Enable the beta tester feature
- Go to Dashboard > Updates and update to the latest beta version.
- This will get you set up to test the new Elementor 3.14.0 - Beta 4.

----

## Notes

As we make changes to our GitHub in an effort to keep it organized, improve the way we deliver information to you, and our ability to review your feedback, please follow the instructions below when reporting bugs, or making feature requests:

- To report a bug related to beta versions - open a new [Bug report](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml)
- For feature requests - open a new [Feature request issue](https://github.com/orgs/elementor/discussions/new?category=feature-request)
- For other comments specifically about this beta version - comment here

Designated time for release: `June 19, 2023 (Estimated)`

Many thanks for your support and help!

Cheers,
_Elementor Team_

**Comment by perks98** [comment]
Got a few quick questions about usage restrictions. Will there be a cap per day or can you let's say generate and enhance 500 images a day? And the current AI price per month with you is 2.99 USD, will this go up or was that cost already accounted for?

**Reply by nicholaszein** [reply]
Hi @perks98,

thanks for showing interest in this new feature. I'll answer your questions, but please remember that we are still in the beta phase, and some of your questions don't have answers yet or may change in the near future. We will have more concrete answers after the product is released.

1. **Q:** Usage restrictions - are there daily limits?
    **A:** No, there are no daily limits. You can generate as many images as you want until you reach your plan's limit.
2. **Q:** Will the price increase with the new AI Image, or was that cost already accounted for?
    **A:** In Elementor 3.14 - beta 4, Elementor AI was expanded to include image generation, and this has automatically been added to your subscription at no extra cost.

**Reply by perks98** [reply]
@nicholaszein thanks for the info.

**Reply by perks98** [reply]
@nicholaszein it seems the pricing has just changed recently and now it has gone from unlimited to credit... so there is a limit of 540 images you can create **per year**.

![Screenshot_815](https://github.com/elementor/elementor/assets/55829113/38b91045-6808-4a44-9eca-beeac03d15eb)

<hr>

Can your team consider creating a cheaper model e.g. 1.99 USD for just the text AI model and remove image prompts from that as 33 credits can get run dry quickly and bring back the unlimited text prompt usage, especially if our clients can use it over 500+ website accounts.

**Reply by fabigrafo** [reply]
> @nicholaszein it seems the pricing has just changed recently and now it has gone from unlimited to credit... so there is a limit of 540 images you can create **per year**.
> 
> ![Screenshot_815](https://user-images.githubusercontent.com/55829113/245723359-38b91045-6808-4a44-9eca-beeac03d15eb.jpg)
> 
> Can your team consider creating a cheaper model e.g. 1.99 USD for just the text AI model and remove image prompts from that as 33 credits can get run dry quickly and bring back the unlimited text prompt usage, especially if our clients can use it over 500+ website accounts.

Okay, so that's a red flag for me.

**Reply by nicholaszein** [reply]
Hi @perks98 Ill pass it to the AI team. Thanks for the feedback! 🙏

**Comment by perks98** [comment]
An issue I have found is that when enhancing an image e.g. expand image option it saves the new image as a smaller version. For example 1536 by 2048 pixels original image, go in to expand image so it is now 16:9 landscape so it fills in the sides, it then saves the new image as 640 by 359 pixels in the media gallery.

**Reply by nicholaszein** [reply]
Thanks for the feedback @perks98! 🙏 

We will take a look at this! 😉

**Comment by Michaelgimii** [comment]
There are two items in the ui of some sections.

1- Close and upgrade buttons are not in the desired place.

![AI](https://github.com/elementor/elementor/assets/88034902/bd016ec1-af83-4d63-b70a-5ee45a7895f3)


2- In the background section of the container, the responsive icon is not in the desired location.

![responsive](https://github.com/elementor/elementor/assets/88034902/47e47207-deb9-4cdc-baaf-0dd242fe29c8)

**Reply by nicholaszein** [reply]
Hey @Michaelgimii!

Thanks for your feedback! We will definitely take a look at these. 🙏 

Please keep in mind that as part of our continuing efforts to improve the Elementor Editor UI and make it more intuitive and accessible, we'll be looking into how we can enhance it in future updates. Item 2 might fall into this effort, and might not be fixed in time for the release. We can't give a timeframe for when we'll sort this out just yet but know that we're on it. Thanks for your understanding!

**Comment by perks98** [comment]
Quick test this morning, when using the image AI tool on a background it continuously loads, while an image widget is fine. First noticed as I waited 4 minutes before giving up. I've included an example video below if that helps.

https://github.com/elementor/elementor/assets/55829113/bbe57e48-2491-4e14-8b48-a44d6c6c01c3

**Reply by nicholaszein** [reply]
Hi @perks98,

Thanks for finding this one! We're on it! Will be fixed soon! 💪

**Comment by LensDigitalUK** [comment]
A quick question, who owns the copyright of any images generated through Elementor using AI?

**Reply by lukecav** [reply]
If it uses DALL-E 2, the copyright would be with the user.
https://help.openai.com/en/articles/6425277-can-i-sell-images-i-create-with-dall-e

**Reply by nicholaszein** [reply]
HI, you might want to check the FAQ for Elementor AI in this link:
https://elementor.com/help/faq-elementor-ai/

I'll quote it directly:
> Are there any copyright issues with Elementor AI?
> 
> Refer to our terms and conditions for more information about copyright and Elementor AI.

**Reply by LensDigitalUK** [reply]
So reading that, there are lots of copyright issues from using Elementor AI for client projects.

Something not to be used with clients.

**Reply by nicholaszein** [reply]
Hi @LensDigitalUK,

can you point out what issues you are encountering?

**Comment by fabigrafo** [comment]
I have to say, I'm not a fan of it. I like to be taught a better one, but now I don't see any use for myself personally. Especially as a photographer, I already have a lot of picture material myself and even if I want/have to change something, I have "better" tools like Photoshop. What is the advantage for me if I use this function?

**Reply by nicholaszein** [reply]
Hello Fabien,

I'll quote the [FAQ] (https://elementor.com/help/faq-elementor-ai/) page, which delivers some pretty relevant reasons:

> **Convenience** – Elementor AI is natively integrated within Elementor, meaning you don’t have to go back and forth between the Editor and the AI tool you use, which also accelerates your productivity.
> **Relevant Suggestions** – Anytime you open Elementor AI, you are provided with a number of relevant suggestions, based on the asset you would like to use AI on. For example, if you choose a heading widget and want to use Elementor AI to help write your heading, a number of appropriate suggestions are added to the modal. This empowers you to fine-tune your request and brings you closer to your desired outcome.
> **Context**– Elementor AI is built specifically for Elementor. When writing a request, the result provided already includes the relevant context and specific selectors.

I hope this helps.

**Comment by Psychopomp777** [comment]
Sorry, probaly it's mention elsewhere, but on which "platform" is the image AI based in Elementor?? Midjourney, Firefly DALE etc.? Thanks!

**Reply by lukecav** [reply]
Most likely, it uses DALLE-2 from OpenAI.
https://openai.com/dall-e-2

**Reply by nicholaszein** [reply]
Hello @Psychopomp777, and @lukecav.

Elementor uses a number of technologies offered by trusted partners. The combination of these technologies is what makes Elementor AI. For now, we're not sharing specifics about these tools. Thanks for understanding.

**Reply by Psychopomp777** [reply]
Honestly: that's a very strange answer and another reason to switch over to another page builder! Bye!

**Comment by bkjproductions** [comment]
What is the impact on the size of the Elementor plugin, what with all these fancy new AI additions? We're concerned with these new features being added, as most of our designs are not done "on the fly" using Elementor, we have a different creative process that doesn't need any AI stuff, as the copy has to be approved before the designer gets into it, there are various approvals etc.

**Reply by LensDigitalUK** [reply]
100% agree with this.

**Reply by lukecav** [reply]
The zip size has increased maybe 1MB or so. Elementor 3.12.2 is 7.9 MB in size whilst Elementor 3.14.0 beta4 is 8.9 MB in size.

**Reply by nicholaszein** [reply]
There is zero impact on the plugin. Since the addition of AI, other features were added too, and measuring plugin zip file sizes does not account for those other features.

It seems what you're asking is that if having the AI feature will impact the usage of the plugin performance wise. The answer is no. The plugin will remain working in the same way. The AI feature loads dynamically and only when needed. When you click on the icon you'll see it starts loading. This means that it doesn't matter if you choose to use it or not. It won't affect the performance of the builder or your workflow if you decide not to use it.

**Reply by LensDigitalUK** [reply]
But it does have impact on the plugin as you are introducing code which is accessing an external service.

As @bkjproductions said, most workflows have all of the images and copy created before building the website so most creative processes don't need AI at the point of creation. They need them in the design phase which is before Elementor. 

This is a tool that is needed in Figma / Canva, not in Elementor.

**Reply by bkjproductions** [reply]
We DEFINITELY need a way to turn off the AI feature for most of our clients. It would serve only to confuse them. :)

**Reply by bkjproductions** [reply]
> The zip size has increased maybe 1MB or so. Elementor 3.12.2 is 7.9 MB in size whilst Elementor 3.14.0 beta4 is 8.9 MB in size.

Unzipped, over 6MB more:
46.8 MB version 3.9.2
40.0 MB  version 3.13.2

**Reply by nicholaszein** [reply]
Thanks for the feedback guys! We appreciate it! 🙏

**Comment by perks98** [comment]
I've just tried this new image AI tool a handful of times now and can say the results are bad, I have tried DALLE-2 (OpenAI's image tool) before and that was my conclusion months ago, rarely for me it does a good job. You can create a lot of cursed images if you add people or hands and some objects it can create for a background can be smeary or morphed. But tend to get better results for background abstract art images. With generative fill, it seems to not use the image as a reference properly and just creates its own thing that sometimes doesn't even blend with the image. Just my thoughts. Unfortunately the company Elementor account I use seems to have run out of AI uses so I can't comment further on this new trial.

**Reply by nicholaszein** [reply]
Thanks for the feedback @perks98, we really appreciate it! 🙏

Elementor AI is ~~specifically trained for Elementor, and we are just starting~~ integrated with the Elementor Editor to give you relevant suggestions, based on the context, in a convenient package where you don't need to leave the builder to get a similar type of content from an external source. 

As time passes I'm sure the quality and accuracy of our AI features will only increase as the model we are using becomes better at what it does. 

We understand not all results are going to be accurate or realistic right out of the bat depending on the prompt, and you might receive undesired results. For this reason we suggest trying one of the prompts available in our [Prompt Ideas page](https://elementor.com/help/elementor-ai-prompt-bank/), or using one of them as inspiration for your own prompts in order to find comparable results. We will be updating this page with more ideas in the future, especially for image generation.

Thanks for contributing! 🙏


[Edit: I mistakenly suggested the AI was trained for Elementor. This is innacurate. The Elementor AI understands context and can generate content based on your prompts. Apologies for the confusion.]

**Comment by black-eye** [comment]
Please, please, make this AI stuff OPTIONAL, give us an option to turn it off.

Reasons:
1. **It's still an beta experiment.** I don't want to be an involuntary beta tester on production sites. Well, in fact, I personally don't care, but I don't want you to use my clients as these unvoluntary betatesters. And yes, publishing it for the first time in a Beta 4 a week before the final release isn't fortunate either...
2. Since it's still in a beta phase, it **messes up the UI** as some already pointed out here.
3. My clients work with Elementor a lot as the editors (I prepare the design, they edit it further). The images are part of my design, **I don't want my clients to be encouraged and tempted to use this tool to ruin the layout.**
4. I don't want you to **urge my clients to create another Elementor account** with a (hidden) intention to charge more money. It's basically another ad for extra paid service.
5. It adds yet **another requirement for the sources** to be loaded. And please don't tell me it's completely harmless to the server resources. If I had a dime for every time I have to request more server memory from the webhost company because of the dreaded spinning wheel...

In fact, I don't expect this request to be satisfied, because you always do, what you think is the best (for you). But I just had to try... ;-)

**Reply by LensDigitalUK** [reply]
I will back you up on this. AI should not be part of the main Elementor plugin. It should be an extra plugin.

With the legal state of AI in flux around the world, I have clients that would be unhappy with me installing an AI plugin on their website. I don't want to have to stop using Elementor but adding more and more AI into it is something that will push me away from it.

**Reply by perks98** [reply]
I agree.

I think it should all be disabled by default so it removes the AI indicator on image and text boxes, then when manually enabled in the experiments/features panel it should prompt you to accept any privacy notice and add restrictions on user access from there for just that site, so when enabled it can then show the AI indicator in the editor.

**Reply by bkjproductions** [reply]
@perks98  logic checks out,
Until then:

`button.e-ai-button {display: none;}`

:)

**Reply by nicholaszein** [reply]
Hello @black-eye,

Please keep in mind that this feature, like any other Elementor feature is optional. You don't need to use it if you don't want to, and it won't impact in any way your workflow or editor performance.

As for the UI conflicts, those have been addressed already, so you don't need to be concerned about it too 🙂. 

We understand your concerns and we will discuss them internally.

Thank you for your honest feedback! 🙏

**Reply by black-eye** [reply]
@nicholaszein Well the the only option I have now is to ignore the function exists (or to use the `button.e-ai-button {display: none;}` hack).

Please understand that there are different use cases and workflows and it would be much more efficient if you gave us, the users, the option to tweak the tool we use to serve it the best way for our usecases. TBH I really don't understand Elementor's reluctance to give us such options (the same with the possibility to turn off some widgets and features).

But as I wrote earlier, I didn't expect this request to resonate well with your plans. ;-) I just couldn't help to write down some arguments, which I found valuable.

Thanks anyway.

**Reply by nicholaszein** [reply]
Thank you for the feedback @black-eye! We really appreciate it! 

We'll surely discuss this internally. 🙏

**Comment by mateitudor** [comment]
Please. With all the respect in the world. Please fix the bugs. We just want a usable product. :(

**Comment by nicholaszein** [comment]
Hello, everyone!

### 📢 Elementor AI Image has been released with **version 3.14.0**! 🥳 

✅ Feel free to check it out and update your plugins to the new version!

Cheers 🥂

> Changelogs of Elementor Core and Pro
> [Elementor Core](https://wordpress.org/plugins/elementor/#developers)
> [Elementor Pro](https://elementor.com/pro/changelog/)
